src/Badges.sol

```diff
@@ -35,8 +39,20 @@ contract Badges is
 
   mapping(uint256 => uint256) private voucherHashIds;
 
+  mapping(uint256 => bool) public revokedBadges;
+  mapping(uint256 => uint256) public badgeExpirationDates;
```

Comment:
Have you considered using block numbers? Just an idea..
https://eips.ethereum.org/EIPS/eip-5081

Reply:
nope! checking it out

Reply:
After reading through it I'm not sure how we could use that EIP here. 

Are you thinking that instead of mapping a `badgeId` to a timestamp that we would map a `badgeId` to `block.number`?

Reply:
Yes. But it's just an idea. I'm not yet sure if it's the best approach. If we end up using blockNumbers, we'd need to compute and estimate what the blocknumber could be at a given point in time. Using blockNumber seems to be the norm in general but from a UX standpoint, this maybe a no-go. Not sure just yet.